st2common/st2common/models/api/base.py

```diff
@@ -245,16 +243,6 @@ def cast_func(value):
                     except Exception as e:
                         raise exc.HTTPInternalServerError(detail=e.message,
                                                           comment=traceback.format_exc())
-
-                    # Set default pack if one is not provided for resource create
-                    if function_name == 'post' and not hasattr(obj, 'pack'):
-                        extra = {
-                            'resource_api': obj,
-                            'default_pack_name': DEFAULT_PACK_NAME
-                        }
-                        LOG.debug('Pack not provided in the body, setting a default pack name',
-                                  extra=extra)
-                        setattr(obj, 'pack', DEFAULT_PACK_NAME)
```

Comment:
Yeah, I never liked it, it was hacky as hell.

I think we rely on it for actions and rules.